src/server/auth/server/api_server.go

```diff
@@ -1477,70 +867,17 @@ func (a *apiServer) GetAuthToken(ctx context.Context, req *auth.GetAuthTokenRequ
 
 // GetAuthToken implements the protobuf auth.GetAuthToken RPC
 func (a *apiServer) GetAuthTokenInTransaction(txnCtx *txnenv.TransactionContext, req *auth.GetAuthTokenRequest) (resp *auth.GetAuthTokenResponse, retErr error) {
-	if !a.isActive() {
+	if err := a.isActive(); err != nil {
 		// GetAuthToken must work in the partially-activated state so that PPS can
 		// get tokens for all existing pipelines during activation
-		return nil, auth.ErrNotActivated
-	}
-
-	// Get current caller and authorize them if req.Subject is set to a different
-	// user
-	callerInfo, err := a.getAuthenticatedUser(txnCtx.ClientContext)
-	if err != nil {
-		return nil, err
-	}
-	isAdmin, err := a.hasClusterRole(txnCtx.ClientContext, callerInfo.Subject, auth.ClusterRole_SUPER)
-	if err != nil {
 		return nil, err
 	}
 
-	if req.TTL < 0 && !isAdmin {
-		return nil, errors.Errorf("GetAuthTokenRequest.TTL must be >= 0")
-	}
-
-	// check if this request is auhorized
-	req.Subject, err = a.authorizeNewToken(txnCtx.ClientContext, callerInfo, isAdmin, req.Subject)
-	if err != nil {
-		if errors.As(err, &auth.ErrNotAuthorized{}) {
-			// return more descriptive error
-			return nil, &auth.ErrNotAuthorized{
-				Subject: callerInfo.Subject,
-				AdminOp: "GetAuthToken on behalf of another user",
-			}
-		}
-		return nil, err
-	}
-
-	// TODO: switch the ppsUser to have the `pach:` prefix to simplify this case
-	if req.Subject == ppsUser || strings.HasPrefix(req.Subject, auth.PachPrefix) {
+	if strings.HasPrefix(req.Subject, auth.PachPrefix) {
 		return nil, errors.Errorf("GetAuthTokenRequest.Subject is invalid")
 	}
 
-	// Compute TTL for new token that the user will get once OTP is exchanged
-	// Note: For Pachyderm <1.10, admin tokens always come with an indefinite
-	// session, unless a limit is requested. For Pachyderm >=1.10, Admins always
-	// use default TTL (30 days currently), unless they explicitly request an
-	// indefinite TTL.
-	//
-	// TODO(msteffen): Either way, admins can can extend their session by getting
-	// a new token for themselves. I don't know if allowing users to do this is
-	// bad security practice (it might be). However, preventing admins from
-	// extending their own session here wouldn't improve security: admins can
-	// currently manufacture a buddy account, promote it to admin, and then extend
-	// their session indefinitely with the buddy account. Pachyderm cannot prevent
-	// this with the information is currently stores, so preventing admins from
-	// extending their session indefinitely would require larger changes to the
-	// auth model.
-	if !isAdmin {
-		// Caller is getting OTP for themselves--use TTL of their current token
-		ttl, err := a.getCallerTTL(txnCtx.ClientContext)
```

Comment:
Just checking: is this getting removed? Can users still get an auth token for themselves, or is this locked to admins now?

TBH If users can get auth tokens for themselves, I do think that this is valuable from a security perspective (e.g. even though we're unifying auth under OIDC, I think ID tokens can have an `expires_in` constraint that, because we swap the ID token for an access token, it's on us to enforce. Users could work around it, though, if they could get a new token for themselves that had a longer expiration than their current token).

Reply:
The way I'm thinking about auth tokens in 2.0 is: 

- `user` tokens are only available by logging in through the IDP. Nobody, not even admins, can impersonate a real person, and a user can't extend their own session by generating tokens for their own identity. 

- `robot` tokens can be created by admins for things like CI. This is a privileged action because a robot can live beyond the lifetime of a user, even indefinitely.

- `pipeline` tokens can be created by a user if they have appropriate permissions on the pipeline itself. 

In a future PR I'd like to split out `CreateRobotToken` and `CreatePipelineToken` into two separate RPCs so we don't have complicated string parsing - instead the user will provide the robot/pipeline name and we'll build the principal name and validate their access.

Reply:
This makes sense to me now, thanks!

I'm starting to feel like we need a list of product changes in auth 2.0, because there are enough that customer team will need a bit of time and tutoring before they can sell and support it (especially e.g. Joey and JK, who I think are pretty used to auth 1.0 and will need to be deprogrammed a bit).

I don't know what the best place for this to live would be, but to illustrate (and hopefully help), I started just a skeleton here: https://docs.google.com/document/d/1NaRJC_4kPbwSa629_-u8ytb1eZ1qu7uwkt5uImrJ2vQ/edit